src/Components/Shared/src/WebEventData/MouseEventArgsReader.cs

```diff
@@ -0,0 +1,97 @@
+
+        internal static MouseEventArgs Read(JsonElement jsonElement)
+        {
+            var eventArgs = new MouseEventArgs();
+            Read(jsonElement, eventArgs);
+            return eventArgs;
+        }
+
+        internal static void Read(JsonElement jsonElement, MouseEventArgs eventArgs)
+        {
+            foreach (var property in jsonElement.EnumerateObject())
+            {
+                if (property.NameEquals(Detail.EncodedUtf8Bytes))
+                {
+                    eventArgs.Detail = property.Value.GetInt64();
+                }
+                else if (property.NameEquals(ScreenX.EncodedUtf8Bytes))
+                {
+                    eventArgs.ScreenX = property.Value.GetDouble();
+                }
+                else if (property.NameEquals(ScreenY.EncodedUtf8Bytes))
+                {
+                    eventArgs.ScreenY = property.Value.GetDouble();
+                }
+                else if (property.NameEquals(ClientX.EncodedUtf8Bytes))
+                {
+                    eventArgs.ClientX = property.Value.GetDouble();
+                }
+                else if (property.NameEquals(ClientY.EncodedUtf8Bytes))
+                {
+                    eventArgs.ClientY = property.Value.GetDouble();
+                }
+                else if (property.NameEquals(OffsetX.EncodedUtf8Bytes))
+                {
+                    eventArgs.OffsetX = property.Value.GetDouble();
+                }
+                else if (property.NameEquals(OffsetY.EncodedUtf8Bytes))
+                {
+                    eventArgs.OffsetY = property.Value.GetDouble();
+                }
+                else if (property.NameEquals(Button.EncodedUtf8Bytes))
+                {
+                    eventArgs.Button = property.Value.GetInt64();
+                }
+                else if (property.NameEquals(Buttons.EncodedUtf8Bytes))
+                {
+                    eventArgs.Buttons = property.Value.GetInt64();
+                }
+                else if (property.NameEquals(CtrlKey.EncodedUtf8Bytes))
+                {
+                    eventArgs.CtrlKey = property.Value.GetBoolean();
+                }
+                else if (property.NameEquals(ShiftKey.EncodedUtf8Bytes))
+                {
+                    eventArgs.ShiftKey = property.Value.GetBoolean();
+                }
+                else if (property.NameEquals(AltKey.EncodedUtf8Bytes))
+                {
+                    eventArgs.AltKey = property.Value.GetBoolean();
+                }
+                else if (property.NameEquals(MetaKey.EncodedUtf8Bytes))
+                {
+                    eventArgs.MetaKey = property.Value.GetBoolean();
+                }
+                else if (property.NameEquals(Type.EncodedUtf8Bytes))
+                {
+                    eventArgs.Type = property.Value.GetString()!;
+                }
+            }
```

Comment:
This pattern is `O(NM)` with the number of properties on the eventargs type * number of incoming values. I know that it's fixed and end users can't influence that, but this is still a lot of `NameEquals` calls with `N=M=14` here. At what point do you think it would be worth amortizing that by creating a `SortedDictionary<JsonEncodedText, Action<MouseEventArgs, JsonValue>>` or similar up front?

Separately, a hostile user could send an event payload like `{"a":0,"b":0,"c":0,..."zzz":0}` and force a lot of string comparisons per event. I estimate about 57,000 string comparisons per event based on 14 tests per inbound property, and and two-char property name combinations using chars in `a-zA-Z0-9`. Is it worth counting the number of properties we've considered, and throwing if it gets above (say) 50? That wouldn't make the code much more complex.

Reply:
In fact, would it be valid to throw if we see *any* property name that we didn't expect? Our JS client sends a known set, doesn't it (it doesn't just serialize raw JS `Event` objects, does it)?

Reply:
> In fact, would it be valid to throw if we see any property name that we didn't expect?

We could. It'd be different to the current behavior (values without corresponding C# properties get ignored by the serializer) but that might be fine since it would indicate a bug or suggest that we could be sending less data from JS if we genuinely don't need it.

Reply:
Adding a throw for unknown properties to all of the readers doesn't cause any of our tests to fail, so perhaps that's a sufficient solution to limit this sort of attack vector.

Reply:
I'm not sure there's a security concern here, there's a limit to the number of operations you can force and its not controlled by the end user. If we want to make the implementation more "scalable" that's fine, but I don't think we should do anything in the name of security.

There's only an issue when you can cause the work performed by the server to grow without growing your own work. (For example, if you have an input where you enter a number and the server calculates the factorial or fibonacci of that number as an example), but in general it's ok if you have to do more work to cause the server to do more work, even if the scale is different. At that point, its up to the server to establish the appropriate limits of what it considers acceptable.

In this case, even if this can be 57K string comparisons in the worst case, I think it'll be fast enough that it won't matter. It might even be the case, that this way is even faster than doing a hash and a lookup on a Dictionary.

Reply:
I was suggesting a structure that would seek via a series of comparisons, not a hash, but in any case it sounds like we’re all settled on the pattern here with the sequence of Equals checks.